src/plccng/spec/syntax/validations/ll1/check_left_recursion.py

```diff
@@ -0,0 +1,67 @@
+from .Grammar import Grammar
+import copy
+
+def check_left_recursion(grammar: Grammar):
+    return LeftRecursionChecker(grammar).check()
+
+class LeftRecursionChecker:
+    def __init__(self, grammar: Grammar):
+        self.grammar = grammar
+        self.grammarCopy = copy.deepcopy(grammar)
+        self.substitutions = {}
+        self.nonterminalsInOrder = list(self.grammarCopy.getRules().keys())
+        self.rulesDict = self.grammarCopy.getRules()
+
+    def check(self):
+        allLeftRecursionInstances = []
+        for n in self.nonterminalsInOrder:
+            self._handleLeftRecursionSubstitutions(n)
```

Comment:
What does "handle" mean here? How are they "handled"? What does the method do to "handle" them?